### DISCUSSION TITLE
Memory usage

### DISCUSSION THREAD

**Discussion by andyzasl** [martin-helmich/prometheus-nginxlog-exporter#207] [Q&A]
Hi!
I'm using prometheus-nginxlog-exporter-1.8.0-1.x86_64
and found memory usage is about 2G, that is very high for 4Gb system. 
Is there any way to limit usage?

**Comment by martin-helmich** [comment]
In most observations of increased memory consumption, the cause was an increased number of exported metrics, often caused by sub-optimal configuration. Have a look at #199 (in particular, https://github.com/martin-helmich/prometheus-nginxlog-exporter/issues/199#issuecomment-839665623) for more information.

**Comment by andyzasl** [comment]
In my case this is caused by keeping about 500k request_uri strings in memory. 
Is there any reason, why exporter keep it? 
I wonder if there is any setting, like time_to_keep_in_memory, to let exporter free "old" data.
